pop-api/integration-tests/src/fungibles/mod.rs

```diff
@@ -330,122 +304,106 @@ fn token_metadata_works() {
+		// Successfully create a token when instantiating the contract.
+		let result_with_address = instantiate_and_create_fungible(contract, TOKEN_ID, 1);
+		assert_ok!(result_with_address.clone());
+		assert_eq!(Assets::owner(TOKEN_ID), result_with_address.ok());
 		assert!(Assets::asset_exists(TOKEN_ID));
 	});
 }
 
 #[test]
 fn start_destroy_works() {
 	new_test_ext().execute_with(|| {
-		let _ = env_logger::try_init();
 		let addr = instantiate(CONTRACT, INIT_VALUE, vec![2]);
 
-		// Asset does not exist.
-		assert_eq!(start_destroy(addr.clone(), TOKEN_ID), Err(Module { index: 52, error: [3, 0] }),);
-		// Create assets where contract is not the owner.
-		let asset = create_asset(ALICE, 0, 1);
+		// Token does not exist.
+		assert_eq!(start_destroy(&addr, TOKEN_ID), Err(Module { index: 52, error: [3, 0] }),);
+		// Create tokens where contract is not the owner.
+		let token = pallet_assets_create(&ALICE, 0, 1);
 		// No Permission.
-		assert_eq!(start_destroy(addr.clone(), asset), Err(Module { index: 52, error: [2, 0] }),);
-		let asset = create_asset(addr.clone(), TOKEN_ID, 1);
-		assert_ok!(start_destroy(addr.clone(), asset));
+		assert_eq!(start_destroy(&addr, token), Err(Module { index: 52, error: [2, 0] }),);
+		let token = pallet_assets_create(&addr, TOKEN_ID, 1);
+		assert_ok!(start_destroy(&addr, token));
 	});
 }
 
 #[test]
 fn set_metadata_works() {
 	new_test_ext().execute_with(|| {
-		let _ = env_logger::try_init();
 		let name = vec![42];
 		let symbol = vec![42];
 		let decimals = 42u8;
 		let addr = instantiate(CONTRACT, INIT_VALUE, vec![]);
 
-		// Asset does not exist.
+		// Token does not exist.
 		assert_eq!(
-			set_metadata(addr.clone(), TOKEN_ID, vec![0], vec![0], 0u8),
+			set_metadata(&addr, TOKEN_ID, vec![0], vec![0], 0u8),
 			Err(Module { index: 52, error: [3, 0] }),
 		);
-		// Create assets where contract is not the owner.
-		let asset = create_asset(ALICE, 0, 1);
+		// Create token where contract is not the owner.
+		let token = pallet_assets_create(&ALICE, 0, 1);
 		// No Permission.
 		assert_eq!(
-			set_metadata(addr.clone(), asset, vec![0], vec![0], 0u8),
+			set_metadata(&addr, token, vec![0], vec![0], 0u8),
 			Err(Module { index: 52, error: [2, 0] }),
 		);
-		let asset = create_asset(addr.clone(), TOKEN_ID, 1);
-		// Asset is not live, i.e. frozen or being destroyed.
-		freeze_asset(addr.clone(), asset);
+		let token = pallet_assets_create(&addr, TOKEN_ID, 1);
+		// Token is not live, i.e. frozen or being destroyed.
+		pallet_assets_freeze(&addr, token);
 		assert_eq!(
-			set_metadata(addr.clone(), TOKEN_ID, vec![0], vec![0], 0u8),
+			set_metadata(&addr, TOKEN_ID, vec![0], vec![0], 0u8),
 			Err(Module { index: 52, error: [16, 0] }),
 		);
-		thaw_asset(addr.clone(), asset);
+		pallet_assets_thaw(&addr, token);
 		// TODO: calling the below with a vector of length `100_000` errors in pallet contracts
 		//  `OutputBufferTooSmall. Added to security analysis issue #131 to revisit.
```

Comment:
not sure this is an issue, just means pallet-contracts properly exists with values too large.

Reply:
Something to keep in mind though because whenever the developer wants to set metadata and uses a value that is too large. In stead of pallet assets that returns an error it is pallet contracts.

Reply:
Would expect that it might fail with out of gas realistically.